core/geometry/src/serialization/IModelJsonSchema.ts

```diff
@@ -376,42 +376,44 @@ export namespace IModelJson {
   }
 
   /**
-   * Interface for Box (or frustum with all rectangular sections parallel to primary xy section)
-   * * Orientation may be given in any `AxesProp`s way (yawPitchRoll, xyVectors, zxVectors)
-   * * if topX or topY are omitted, each defaults to its baseX or baseY peer.
-   * * `topOrigin` is determined with this priority order:
-   * * * `topOrigin` overrides given `height`
-   * * * on the z axis at distance `height`
-   * * * If both `topOrigin` and `height` are omitted, `height` defaults to `baseX`
+   * Interface for Box (or frustum with all rectangular sections parallel to primary xy section).
+   * * Orientation may be given in any `AxesProps` way (`yawPitchRollAngles`, `xyVectors`, `zxVectors`).
+   * * If `topX` or `topY` are omitted, each defaults to its `baseX` or `baseY` peer.
+   * * If `topOrigin` is given, `height` is unused.
+   * * If `topOrigin` is omitted and `height` is given, `topOrigin` is computed along the z axis at distance `height`.
+   * * If both `topOrigin` and `height` are omitted, `height` defaults to `baseX`, and `topOrigin` is computed as above.
    * @public
    */
   export interface BoxProps extends AxesProps {
     /** Origin of the box coordinate system  (required) */
-    origin: XYZProps;
+    baseOrigin: XYZProps;
```

Comment:
This is a breaking change, application that uses this box now require `baseOrigin` to be provided.

Reply:
That is true, but any application that was providing `origin` would have been broken and failed to produce a box, because the native code that reads this JSON requires a `baseOrigin` property to be provided.

Reply:
Not fixing style guidelines in tests as part of addon PR.

Reply:
Still a breaking change in the sense that the code will no longer compile (even if it was not working before) I dont know the extent of what would entail changing the implementation to make this a non breaking change so the part that sends to the native code "fix" the object before sending (remapping `origin` to `baseOrigin`, and make `baseOrigin` the only and good way of doing it in the  4.0 release...)

Reply:
@dassaf4 after this PR completes please submit a PR to remove the breaking API change.

Reply:
Or I will do it.

Reply:
Sounds good, and thanks @raplemie for pointing this out. @pmconne, if the fix is just a matter of adding logic to IModelJson.Reader.parseBox() to look for an origin property if baseOrigin isn't found, I can give it a go. Otherwise, I'll learn from how you fix it.